lib/Sema/MiscDiagnostics.cpp

```diff
@@ -1581,7 +1577,9 @@ static void diagRecursivePropertyAccess(const Expr *E, const DeclContext *DC) {
     }
 
     MacroWalking getMacroWalkingBehavior() const override {
-      return MacroWalking::Expansion;
+      // Macro expansions will be walked when they're type-checked, not as
+      // part of the surrounding code.
+      return MacroWalking::None;
```

Comment:
Could you please separately evaluate whether it would be possible to subtype all of this walkers from `BaseDiagnosticWalker` and remove all of these overrides?

Reply:
Heh, I have that exact change in a follow-up PR :)

Reply:
Perfect! :)